### ISSUE TITLE
Use of Apple's Required Reason apis requires privacy manifest

### ISSUE THREAD

**Issue by Rehubbard** [microsoft/react-native-code-push#2679]
Apple is requiring packages and apps to declare their use of required reason apis. This library uses these apis that require a privacy manifest declaration: 1. NSFileModificationDate 2.NSUserDefaults

More info here: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api

Beginning May 1, 2024, Apple won't accept new apps or updates to existing apps that use packages that don't have a `PrivacyInfo.xcprivacy`.

**Comment by hapablap21** [comment]
Also NSFileCreation date (along with NSFileModificationDate) are referenced in the SSZipArchive package, which appears to just be copy/pasted into the RNCP/ios folder.  SSZipArchive's repo has been updated to comply as of last October (https://github.com/ZipArchive/ZipArchive/issues/687) but the version in this package needs to be updated.

**Comment by vomchik** [comment]
Any update on this, folks? It's blocker after May 1 for release process

**Comment by MinskLeo** [comment]
Same for me. Would be good to have this manifest included until 1 of May...

**Comment by julian-dotcom** [comment]
Any news on this?

**Comment by vomchik** [comment]
You can apply this patch. 

https://www.npmjs.com/package/patch-package 

```
diff --git a/node_modules/react-native-code-push/CodePush.podspec b/node_modules/react-native-code-push/CodePush.podspec
index d20c4f4..bee88a3 100644
--- a/node_modules/react-native-code-push/CodePush.podspec
+++ b/node_modules/react-native-code-push/CodePush.podspec
@@ -21,7 +21,7 @@ Pod::Spec.new do |s|
   # we explicitly let CocoaPods pull in the versions below so all dependencies are resolved and 
   # linked properly at a parent workspace level.
   s.dependency 'React-Core'
-  s.dependency 'SSZipArchive', '~> 2.2.2'
+  s.dependency 'SSZipArchive', '~> 2.5.5'
   s.dependency 'JWT', '~> 3.0.0-beta.12'
   s.dependency 'Base64', '~> 1.1'
 end
\ No newline at end of file
```

**Comment by fxamauri** [comment]
Hi everyone, has anyone applied this patch yet? Did it work?

**Comment by fxamauri** [comment]
> You can apply this patch.
> 
> https://www.npmjs.com/package/patch-package
> 
> ```
> diff --git a/node_modules/react-native-code-push/CodePush.podspec b/node_modules/react-native-code-push/CodePush.podspec
> index d20c4f4..bee88a3 100644
> --- a/node_modules/react-native-code-push/CodePush.podspec
> +++ b/node_modules/react-native-code-push/CodePush.podspec
> @@ -21,7 +21,7 @@ Pod::Spec.new do |s|
>    # we explicitly let CocoaPods pull in the versions below so all dependencies are resolved and 
>    # linked properly at a parent workspace level.
>    s.dependency 'React-Core'
> -  s.dependency 'SSZipArchive', '~> 2.2.2'
> +  s.dependency 'SSZipArchive', '~> 2.5.5'
>    s.dependency 'JWT', '~> 3.0.0-beta.12'
>    s.dependency 'Base64', '~> 1.1'
>  end
> \ No newline at end of file
> ```


Hi everyone, has anyone applied this patch yet? Did it work?

**Comment by bunnings-kamrul** [comment]
I have tried this getting error when I try to install pod 

```
[!] CocoaPods could not find compatible versions for pod "SSZipArchive":
  In snapshot (Podfile.lock):
    SSZipArchive (= 2.2.3, ~> 2.2.2)

  In Podfile:
    CodePush (from `../node_modules/react-native-code-push`) was resolved to 8.2.2, which depends on
      SSZipArchive (~> 2.5.5)


You have either:
 * changed the constraints of dependency `SSZipArchive` inside your development pod `CodePush`.
   You should run `pod update SSZipArchive` to apply changes you've made.
```

When I try to update `SSZipArchive`, I get following issue

```
[!] CocoaPods could not find compatible versions for pod "SSZipArchive":
  In Podfile:
    CodePush (from `../node_modules/react-native-code-push`) was resolved to 8.2.2, which depends on
      SSZipArchive (~> 2.5.5)

Specs satisfying the `SSZipArchive (~> 2.5.5)` dependency were found, but they required a higher minimum deployment target.

```

I have tried to update minimum ios version to `15.5` by setting 
`platform :ios, '15.5'` 
in the `Podfile` without any luck. 

Anyone has any luck.

**Comment by bunnings-kamrul** [comment]
There are two places the minimum iOS version to `15.5` need to be updated, one in the local Podfile and also in the code-push podspec. Here is the updated podspec. 

```
diff --git a/node_modules/react-native-code-push/CodePush.podspec b/node_modules/react-native-code-push/CodePush.podspec
index d20c4f4..6718c90 100644
--- a/node_modules/react-native-code-push/CodePush.podspec
+++ b/node_modules/react-native-code-push/CodePush.podspec
@@ -10,7 +10,7 @@ Pod::Spec.new do |s|
   s.license        = package['license']
   s.homepage       = package['homepage']
   s.source         = { :git => 'https://github.com/microsoft/react-native-code-push.git', :tag => "v#{s.version}"}
-  s.ios.deployment_target = '9.0'
+  s.ios.deployment_target = '15.5'
   s.tvos.deployment_target = '9.0'
   s.preserve_paths = '*.js'
   s.library        = 'z'
@@ -21,7 +21,7 @@ Pod::Spec.new do |s|
   # we explicitly let CocoaPods pull in the versions below so all dependencies are resolved and 
   # linked properly at a parent workspace level.
   s.dependency 'React-Core'
-  s.dependency 'SSZipArchive', '~> 2.2.2'
+  s.dependency 'SSZipArchive', '~> 2.5.5'
   s.dependency 'JWT', '~> 3.0.0-beta.12'
   s.dependency 'Base64', '~> 1.1'
 end

```

This got the package working for me at least.

**Comment by MikhailSuendukov** [comment]
Hello, I would like to clarify whether the solution described above works for you?
If it doesn’t, I recommend a temporary workaround: manually adding a Privacy Info file, similar to the one in the SSZipArchive package.

**Comment by ujeon** [comment]
@MikhailSuendukov Hello,

>manually adding a Privacy Info file, similar to the one in the SSZipArchive package.

Should I copy and paste [PrivacyInfo](https://github.com/ZipArchive/ZipArchive/blob/be6ac340cbe394cc6a81eae6034a91a7702d9db3/SSZipArchive/Supporting%20Files/PrivacyInfo.xcprivacy#L5) of ZipArchive to react-native-code-push and apply patch-package?

**Comment by MikhailSuendukov** [comment]
@ujeon 
Yes, you can try doing this as a temporary workaround.

**Comment by MikhailSuendukov** [comment]
A fix for this issue was released with the AppCenter SDK ReactNative v9.0.0, so I'm closing this issue.
